src/pages/instances/actions/snapshots/DuplicateInstanceSnapshotBtn.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC } from "react";
+import { LxdInstance, LxdInstanceSnapshot } from "types/instance";
+import { Button, Icon } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import DuplicateInstanceSnapshotForm from "../../forms/DuplicateInstanceSnapshotForm";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  isDeleting: boolean;
+  isRestoring: boolean;
+}
+
+const DuplicateInstanceSnapshotBtn: FC<Props> = ({
```

Comment:
I would call this `CreateInstanceFromSnapshot`, wdyt?

Reply:
Yes agreed, especially since we already have the `CreateInstanceFromImageBtn` component, better to keep naming consistent.